compiler/rustc_privacy/src/lib.rs

```diff
@@ -476,7 +507,7 @@ impl<'tcx> EmbargoVisitor<'tcx> {
             return;
         }
 
-        if self.get(local_def_id).is_none() {
+        if self.effective_visibilities.public_at_level(local_def_id).is_none() {
```

Comment:
Macros 2.0 that are not `pub`, but e.g. `pub(crate)` can increase items' reachability inside the crate too.
But this can be fixed later, macro reachability needs many fixes and moving to `rustc_resolve` anyway.